### PR TITLE
feat: Add linter and prettier

### LINKED ISSUE #39: Add linter and prettier to the new service repo
@kevinszuchet commented on [Thu Feb 16 2023](https://github.com/decentraland/decentraland-dapps/issues/373)

To complete this task, the following must be done:
- Add the lint and prettier mechanisms to prettify the code.
- Add linter checks to the CI.

### PR DESCRIPTION
**Pull request by kevinszuchet**
Closes #39

### CODE REVIEW

**Review by LautaroPetaccio — APPROVED** [review]

File: .husky/pre-commit [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+
+echo "Running prettier"
+npm run pre-commit:fix:prettier -- $FILES
+
+TS_FILES=$(git diff --cached --name-only --diff-filter=ACMR | sed 's| |\\ |g' | { grep -E '(js|ts|tsx)$' || true; })
+
+if [[ ! -z "$TS_FILES" ]];then
+    echo "Running lints"
+    npm run pre-commit:fix:code -- $TS_FILES
+fi
```
Comment (cyaiox) [inline .husky/pre-commit:18]: missing eol

File: package.json [hunk 2]
```diff
@@ -7,27 +7,31 @@
...
+    "pre-commit:fix:prettier": "prettier --write",
+    "prepare": "husky install"
   },
   "devDependencies": {
+    "@dcl/eslint-config": "^1.1.1",
     "@types/jest": "^29.5.0",
     "@types/node": "^16.18.4",
     "@well-known-components/test-helpers": "^1.4.0",
-    "dcl-tslint-config-standard": "^3.0.0",
+    "eslint-plugin-prettier": "^4.2.1",
```
Comment (cyaiox) [inline package.json:25]: np: I think this package is not required

File: src/controllers/handlers/picks-handlers.ts [hunk 3]
```diff
@@ -1,43 +1,43 @@
...
   const userAddress: string | undefined = verification?.auth.toLowerCase()
 
   try {
-    const power = getNumberParameter("power", url.searchParams)
+    const power = getNumberParameter('power', url.searchParams)
 
-    const pickStats = await picks.getPickStats(params.itemId, { userAddress, power: power ?? undefined })
+    const pickStats = await picks.getPickStats(params.itemId, {
+      userAddress,
+      power: power ?? undefined
```
Comment (cyaiox) [inline src/controllers/handlers/picks-handlers.ts:23]: the method `getNumberParameter` returns `undefined` when it doesn't find a query param. 
```suggestion
      power
```

File: src/migrations/1677778846950_lists-and-picks.ts [hunk 4]
```diff
@@ -1,44 +1,60 @@
...
 
-const PICKS_TABLE = "picks"
-const LISTS_TABLE = "lists"
+const PICKS_TABLE = 'picks'
+const LISTS_TABLE = 'lists'
 
 export const shorthands: ColumnDefinitions | undefined = undefined
 
+// TODO: handle the following eslint-disable statement
+// eslint-disable-next-line @typescript-eslint/require-await
```
Comment (cyaiox) [inline src/migrations/1677778846950_lists-and-picks.ts:10]: if this migration is not doing async operations, why return a promise?

**Review by cyaiox — APPROVED** [review]
LGTM, left a few comments but approved. Good job 🚀

File: src/migrations/1677778846950_lists-and-picks.ts (hunk 4 above)
Comment (LautaroPetaccio) [reply src/migrations/1677778846950_lists-and-picks.ts:10]: I think we could remove the async from the migrations and it should work fine, right? AFAIK, the migration tool creates the functions with the async keyword, but it's not really necessary.

Comment (cyaiox) [reply src/migrations/1677778846950_lists-and-picks.ts:10]: yep, it should work fine

File: package.json [hunk 5]
```diff
@@ -7,27 +7,31 @@
     "start:watch": "nodemon src/index.ts",
     "migrate": "node-pg-migrate --database-url-var PG_COMPONENT_PSQL_CONNECTION_STRING --envPath .env -j ts --tsconfig tsconfig.json -m ./src/migrations",
     "test": "jest --forceExit --detectOpenHandles --coverage --verbose",
-    "test:watch": "jest --watch --verbose"
+    "test:watch": "jest --watch --verbose",
+    "lint": "eslint -c .eslintrc.json --ext .ts src",
+    "check:code": "eslint -c .eslintrc.json src",
+    "check:prettier": "prettier -c 'src/**/*.{js,ts,json,yml,md}'",
```
[the remark this answers is not in the document]
Comment (cyaiox) [reply package.json:13]: the extended eslint config already validates the prettier rules. 

In this case, `lint` and `check:code` do the same. we could remove `check:code` and rename `check:prettier` to `lint:prettier`. wdyt?

**Pull request merged**